tests/storage/moosefs/test_moosefs_sr.py

```diff
@@ -0,0 +1,92 @@
+import pytest
+from lib.common import wait_for, vm_image
+import time, subprocess
+
+# Requirements:
+# - one XCP-ng host >= 8.2
+# - remote moosefs mountpoint
+# - access to XCP-ng RPM repository from the host
+
+class TestMooseFSSRCreateDestroy:
+    """
+    Tests that do not use fixtures that setup the SR or import VMs,
+    because they precisely need to test SR creation and destruction,
+    and VM import.
+    """
+
+    def test_create_moosefs_sr_without_mfsmount(self, host, moosefs_device_config):
+        # This test must be the first in the series in this module
+        assert not host.file_exists('/usr/sbin/mount.moosefs'), \
+            "mount.moosefs must not be installed on the host"
+        sr = None
+        try:
+            sr = host.sr_create('moosefs', "MooseFS-SR-test", moosefs_device_config, shared=True)
+        except Exception:
+            print("MooseFS SR creation failed, as expected.")
+        if sr is not None:
+            sr.destroy()
+            assert False, "MooseFS SR creation should failed!"
+
+    def test_create_and_destroy_sr(self, host_with_moosefs, moosefs_device_config):
+        # Create and destroy tested in the same test to leave the host as unchanged as possible
+        host = host_with_moosefs
+        sr = host.sr_create('moosefs', "MooseFS-SR-test", moosefs_device_config, shared=True, verify=True)
+        # import a VM in order to detect vm import issues here rather than in the vm_on_xfs_fixture used in
+        # the next tests, because errors in fixtures break teardown
+        vm = host.import_vm(vm_image('mini-linux-x86_64-bios'), sr_uuid=sr.uuid)
```

Comment:
As for other storage tests can you use the `--vm` option and not always use `'mini-linux-x86_64-bios'`?

Reply:
The vm param is given to functions like `vm_on_XXX_sr`, it's ok to use a hardcoded VM image here. (It's already the case in other driver tests.)

Reply:
I have just make a commit connected with this thread: f3d3f2b
Please check if this type of solution meets your expectations.

Reply:
Like I said you can keep the previous code with the hardcoded 'mini-linux-x86_64-bios' VM, because we just want to test quickly the SR creation like in other drivers. :wink: It's sometimes useful to use a specific image using the CLI but not in this test.